nexus3/pkg/repository/common/service.go

```diff
@@ -22,11 +23,16 @@ func NewRepositoryService[R any](ep string, c *client.Client) *RepositoryService
 }
 
 func (s *RepositoryService[R]) Create(repo R) error {
+	return s.CreateContext(context.Background(), repo)
```

Comment:
@anmoel it is shorter than duplicate code with `CreateContext` like this:

```go
func (s *RepositoryService[R]) CreateContext(ctx context.Context, repo R) error {
	data, err := tools.JsonMarshalInterfaceToIOReader(repo)
	if err != nil {
		return err
	}
	body, resp, err := s.client.Post(s.endpoint, data)
	if err != nil {
		return err
	}
	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
		return fmt.Errorf("could not create repository %v: HTTP: %d, %s", repo, resp.StatusCode, string(body))
	}
	return nil
}
```

Reply:
i mean, why you want to manage the context? in our old code, we don't manage the context. we used the http client functions without context.

Reply:
To support request cancellation, timeout and so on.

An example use-case is `terraform-provider-nexus`: All `Resource.Create|Delete|Exists|ReadUpdate` methods are deprecated in favor of [`XContext` counterparts](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema@v2.36.1#Resource.Create)
```
// Deprecated: Use CreateContext or CreateWithoutTimeout instead. This
// implementation does not support request cancellation initiated by
// Terraform, such as a system or practitioner sending SIGINT (Ctrl-c).
// This implementation also does not support warning diagnostics.
Create [CreateFunc](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema@v2.36.1#CreateFunc)
```